packages/client/hmi-client/src/model-representation/service.ts

```diff
@@ -189,16 +189,64 @@ export function getParameter(model: Model, parameterId: string): ModelParameter
 	}
 }
 
+export function setParameters(model: Model, parameters: ModelParameter[]) {
+	const modelType = getModelType(model);
+	switch (modelType) {
+		case AMRSchemaNames.REGNET:
+			model.model.parameters = parameters;
+			break;
+		case AMRSchemaNames.PETRINET:
+		case AMRSchemaNames.STOCKFLOW:
+		default:
+			if (model.semantics) model.semantics.ode.parameters = parameters;
+			break;
+	}
+}
+
+export function updateParameter(model: Model, parameterId: string, key: string, value: any) {
+	const parameters = getParameters(model);
+	const parameter = parameters.find((p: ModelParameter) => p.id === parameterId);
+	if (!parameter) return;
+	if (key === 'units') {
+		if (!parameter.units) parameter.units = { expression: '', expression_mathml: '' };
+		parameter.units.expression = value;
+	} else {
+		parameter[key] = value;
+	}
+
+	// FIXME: (For stockflow) Sometimes auxiliaries can share the same ids as parameters so for now both are be updated in that case
+	const auxiliaries = model.model?.auxiliaries ?? [];
+	const auxiliary = auxiliaries.find((a) => a.id === parameterId);
+	if (!auxiliary) return;
+	auxiliary[key] = value;
+}
+
 /**
  * Retrieves the metadata for a specific initial in the model.
  * @param {Model} model - The model object.
- * @param {string} initialId - The ID of the initial.
+ * @param {string} target - The target of the initial.
  * @returns {any} - The metadata for the specified initial or undefined if not found.
  */
 export function getInitialMetadata(model: Model, parameterId: string) {
 	return model.metadata?.initials?.[parameterId];
 }
 
+export function getInitialName(model: Model, target: string): string {
+	return model.metadata?.initials?.[target]?.name ?? '';
+}
+
+export function getInitialDescription(model: Model, target: string): string {
+	return model.metadata?.initials?.[target]?.description ?? '';
+}
+
+export function getInitialUnits(model: Model, target: string): string {
+	return model.metadata?.initials?.[target]?.units?.expression ?? '';
+}
+
+export function getInitialConcept(model: Model, target: string): string {
+	return model.metadata?.initials?.[target]?.concept?.grounding ?? '';
+}
```

Comment:
Why not leverage the above method?

```suggestion
export function getInitialName(model: Model, target: string): string {
	return getInitialMetadata(model, target)?.name ?? '';
}

export function getInitialDescription(model: Model, target: string): string {
	return getInitialMetadata(model, target)?.description ?? '';
}

export function getInitialUnits(model: Model, target: string): string {
	return getInitialMetadata(model, target)?.units?.expression ?? '';
}

export function getInitialConcept(model: Model, target: string): string {
	return getInitialMetadata(model, target)?.concept?.grounding ?? '';
}
```